js/filters/set-filters.ts

```diff
@@ -0,0 +1,28 @@
+import { Photo } from '../contracts/common';
+import { filterOptions, toggleActiveState, filters } from './filters-utils';
+
+const setFilters = (cb: (currentPhotos: Photo[]) => void, photos: Photo[]) => {
+	filters.classList.remove('img-filters--inactive');
+
+	const onFilterButtonClick = (evt: Event) => {
+		const currentTarget = evt.target! as HTMLElement;
+
+		// Если на кнопке два класса, то эта кнопка активна в данный момент и ререндер не нужен
+		const isClickCorrect = (currentTarget.classList[0] !== 'img-filters__button') || (currentTarget.classList.length < 2);
+		if (!isClickCorrect) {
+			return;
+		}
+
+		toggleActiveState(evt);
+		const currentElement = evt.target as HTMLButtonElement;
```

Comment:
Ты же `evt.target` уже вносил в переменную. Если хочешь просто тип поменять, то можно тайпгард написать
```ts
const isClickCorrect = (currentTarget: HTMLElement): currentTarget is HTMLButtonElement => (currentTarget.classList[0] !== 'img-filters__button') || (currentTarget.classList.length < 2);
```
И потом
```suggestion
		if (!isClickCorrect(currentTarget)) {
			return;
		}

		toggleActiveState(evt);
		const currentFilterName = currentTarget.id.split('-')[1] as FilterOption;
```